apps/prairielearn/python/prairielearn.py

```diff
@@ -1475,7 +1507,8 @@ def string_to_2darray(s, *, allow_complex=True):
         # Return result with no error
         return (matrix, {"format_type": "python"})
 
-    assert_never(number_of_left_brackets)
+    # TODO: is there a way to statically check that we've handled all cases?
```

Comment:
I think there is: if you first turn the `number_of_left_brackets` comparisons into an enum, you can then do exhaustiveness checking and `assert_never(...)` on the enum value. e.g.:

```python
    class ResultType(Enum):
        SCALAR = "scalar"
        MATLAB = "matlab"
        PYTHON = "python"

    if number_of_left_brackets == 0:
        result_type = ResultType.SCALAR
    if number_of_left_brackets == 1:
        result_type = ResultType.MATLAB
    else:
        result_type = ResultType.PYTHON

    if result_type == ResultType.SCALAR:
        return ...
    if result_type == ResultType.MATLAB:
        return ...
    if result_type == ResultType.PYTHON:
        return ...
    assert_never(result_type)
```

Reply:
Did a pass in 98f7cbb6d0a54e0daca7c6375a84818af1962d20.